source/extensions/config_subscription/grpc/grpc_mux_impl.h

```diff
@@ -131,7 +149,12 @@ class GrpcMuxImpl : public GrpcMux,
 
   private:
     void updateResources(const absl::flat_hash_set<std::string>& resources) {
-      resources_.clear();
+      // Finding the list of removed resources by keeping the current resources
+      // set until the end the function and computing the diff.
+      // Temporarily keep the resources prior to the update to find which ones
+      // were removed.
+      std::set<std::string> previous_resources;
```

Comment:
Do we do a similar pattern elsewhere (IIRC) for mapping SotW to delta and back? Should we have some smarter resource differencer that can be shared?

Reply:
The delta-xDS/unified-mux use the watch-map [updateWatchInterest](https://github.com/envoyproxy/envoy/blob/1f67c88f26744e1f7c21c0e65798d77d69efeac8/source/extensions/config_subscription/grpc/watch_map.cc#L51) in order to achieve this.
Refactoring it is essentially what the unified-mux has done, as the non-unified SotW doesn't use a watch-map.